libwebauthn/src/fido.rs

```diff
@@ -312,6 +323,13 @@ mod tests {
             &credential_id,
             &webauthn_auth_data[55..55 + &credential_id.len()]
         );
-        assert_eq!(cose_bytes, &webauthn_auth_data[55 + credential_id.len()..]);
+        let extensions_bytes = cbor::to_vec(&extensions).unwrap();
+        assert_eq!(cose_bytes, &webauthn_auth_data[55 + credential_id.len()..webauthn_auth_data.len() - extensions_bytes.len()]);
+
+        let authdata_wrapped = cbor::to_vec(&ByteBuf::from(webauthn_auth_data)).unwrap();
+        let auth_data_reparsed: AuthenticatorData<String> =
+            cbor::from_slice(authdata_wrapped.as_slice()).unwrap();
+        assert_eq!(extensions, auth_data_reparsed.extensions);
```

Comment:
As we already have everything in place here, can you add `assert_eq!(..)` for all the other fields (rp_id_hash, flags, ...) of original vs. reparsed as well?